File: Domain/FeatureToggle.cs
using System;

namespace Domain;

public class FeatureToggle
{
    public string Id { get; set; } = Guid.CreateVersion7(TimeProvider.System.GetUtcNow()).ToString();
    public required string FeatureName { get; set; }

    public bool IsEnabled { get; set; } = true;
    public string? Description { get; set; }

    public DateTime? EffectiveFrom { get; set; }

    public DateTime? EffectiveTo { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public string? UpdatedBy { get; set; }

    public string? Scope { get; set; }
    public string? ScopeValue { get; set; }

    // Navigation property
    public User? Updater { get; set; }
}

File: client/src/lib/types/featureToggle.ts
export interface FeatureToggleDto {
  id: string;
  featureName: string;
  isEnabled: boolean;
  description: string | null;
  effectiveFrom: string | null;
  effectiveTo: string | null;
  scope: string | null;
  scopeValue: string | null;
  createdAt: string;
  updatedAt: string;
  updatedBy: string | null;
}

export interface CreateFeatureTogglePayload {
  featureName: string;
  description?: string | null;
  isEnabled?: boolean;
  scope?: string | null;
  scopeValue?: string | null;
  effectiveFrom?: string | null;
  effectiveTo?: string | null;
}

export interface UpdateFeatureTogglePayload {
  featureName: string;
  description?: string | null;
  isEnabled: boolean;
  scope?: string | null;
  scopeValue?: string | null;
  effectiveFrom?: string | null;
  effectiveTo?: string | null;
}

export interface SetFeatureToggleEnabledPayload {
  isEnabled: boolean;
}

export interface AdminFeatureTogglesQueryParams {
  pageNumber?: number;
  pageSize?: number;
  isEnabled?: boolean | null;
  scope?: string | null;
  search?: string | null;
}

export interface PagedFeatureTogglesResponse {
  items: FeatureToggleDto[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}
